File: clients/node/lib/hashers/hmac-sha256.js
const crypto = require('crypto');

async function hash(value, password) {

    return hashSync(value, password);
}

function hashSync(value, password) {

    return crypto
      .createHmac('sha256', new Buffer(password, 'base64'))
      .update(value)
      .digest("base64");
}

module.exports = {
    hash,
    hashSync
};

File: clients/ruby/lib/ElencyConfig/HMACSHA256.rb
require 'base64'
require 'openssl'

class HMACSHA256

  def hash(value, password)
    hashedValue = Base64.encode64("#{OpenSSL::HMAC.digest('sha256', Base64.decode64(password), value)}")
    return hashedValue.strip
  end

end